platform-http-service-framework/src/main/java/org/hypertrace/core/serviceframework/http/StandAloneHttpPlatformServiceContainer.java

```diff
@@ -0,0 +1,58 @@
+package org.hypertrace.core.serviceframework.http;
+
+import static io.grpc.Deadline.after;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.hypertrace.core.grpcutils.client.GrpcChannelRegistry;
+import org.hypertrace.core.serviceframework.PlatformService;
+import org.hypertrace.core.serviceframework.config.ConfigClient;
+import org.hypertrace.core.serviceframework.http.jetty.JettyHttpServerBuilder;
+
+@Slf4j
+public abstract class StandAloneHttpPlatformServiceContainer extends PlatformService {
+  private HttpContainer container;
+  private final GrpcChannelRegistry grpcChannelRegistry = new GrpcChannelRegistry();
+
+  public StandAloneHttpPlatformServiceContainer(ConfigClient config) {
+    super(config);
+  }
+
+  protected abstract List<HttpHandlerFactory> getHandlerFactories();
+
+  @Override
+  protected void doInit() {
+    this.container =
+        new JettyHttpServerBuilder().addHandlers(this.buildHandlerDefinitions()).build();
+  }
+
+  @Override
+  protected void doStart() {
+    log.info("Starting service {}", this.getServiceName());
+    this.container.start();
+    this.container.blockUntilStopped();
+  }
+
+  @Override
+  protected void doStop() {
+    log.info("Stopping service {}", this.getServiceName());
+    grpcChannelRegistry.shutdown(after(10, SECONDS));
+    this.container.stop();
+  }
+
+  @Override
+  public boolean healthCheck() {
+    return true;
```

Comment:
Nothing more in this one?

Reply:
I resisted 😉 let's file this under the *Future Work* section (it's not a regression and it's still overridable - and more importantly, we didn't really have a ready replacement lined up like we did with GRPC)